service/history/historybuilder/history_builder.go

```diff
@@ -279,13 +279,16 @@ func (b *HistoryBuilder) AddWorkflowTaskFailedEvent(
 func (b *HistoryBuilder) AddActivityTaskScheduledEvent(
 	workflowTaskCompletedEventID int64,
 	command *commandpb.ScheduleActivityTaskCommandAttributes,
+	targetNamespace namespace.Name,
```

Comment:
nit: it's not a "target" namespace, it's just the namespace of the workflow that this event belongs to.
Consider renaming to `ns` or something.